src/main/java/io/aiven/kafka/connect/s3/config/S3SinkConfig.java

```diff
@@ -228,6 +232,72 @@ private static void addAwsConfigGroup(final ConfigDef configDef) {
             ConfigDef.Width.NONE,
             AWS_S3_REGION_CONFIG
         );
+
+        configDef.define(
+                AWS_STS_ROLE_ARN,
+                ConfigDef.Type.STRING,
+                null,
+                new ConfigDef.NonEmptyString(),
+                Importance.MEDIUM,
+                "AWS STS Role",
+                GROUP_AWS,
+                awsGroupCounter++,
+                ConfigDef.Width.NONE,
+                AWS_STS_ROLE_ARN
+        );
+
+        configDef.define(
+                AWS_STS_ROLE_SESSION_NAME,
+                ConfigDef.Type.STRING,
+                null,
+                new ConfigDef.NonEmptyString(),
+                Importance.MEDIUM,
+                "AWS STS Session name",
+                GROUP_AWS,
+                awsGroupCounter++,
+                ConfigDef.Width.NONE,
+                AWS_STS_ROLE_SESSION_NAME
+        );
+
+        configDef.define(
+                AWS_STS_ROLE_SESSION_DURATION,
+                ConfigDef.Type.STRING,
+                "3600",
+                new ConfigDef.NonEmptyString(),
```

Comment:
Should be `INT` or `LONG`. Also, `ConfigDef.Range` as the validator would work.
This will obviate the related checks in `getStsRoleSessionDuration` and `validate`.

Reply:
I wish it was possible. S3SinkTask start function require Map<String, String> as a properties Map.

```
public void start(final Map<String, String> props) {

```

Reply:
It's ok, the whole point of these types is automatic conversion from String, type checks, etc. E.g.: https://github.com/aiven/aiven-kafka-connect-gcs/blob/master/src/main/java/io/aiven/kafka/connect/gcs/GcsSinkConfig.java#L202